### ISSUE TITLE
[SURE-6377] [RFE] Improve steve cache or re-evaluate caching technique.

### ISSUE THREAD

**Issue by rmweir** [rancher/rancher#41378]
Internal reference: [SURE-6377](https://jira.suse.com/browse/SURE-6377)

**Is your feature request related to a problem? Please describe.**
Currently steve implements caching using an LRU cache. It caches a configured number of requests. The problem with this is that it does not consider the size of each request. If the size of a request eclipses that of everything else it still poses a memory risk.

**Describe the solution you'd like**
First an attempt at configuring the amount of data that can be cached by memory. If this is not feasible then reevaluate how caching is being done. If the first part is possible this is more of an enhancement than an RFE.

**Describe alternatives you've considered**
The RFE portion of this issue would be for the dev to explore, consider, design, and get reviewed.

**Additional context**
May add to this later.

**Comment by cbron** [comment]
SURE-6377

**Comment by rmweir** [comment]
Ran benchmarks described here https://github.com/rmweir/scalability-tests/blob/20230927_steve_cache_tests/docs/20230927%20-%20multi%20users%20steve%20cach%20test.md. Both the current steve cache technique and the update [here](https://github.com/rancher/steve/pull/113) use an unreasonable amount of memory in a scaled environment.

[Here](https://github.com/rmweir/steve/tree/steve-cache-rv-update) some adjustments were made that improved the memory usage significantly. However, after reviewing long term goals and assessing that pursuing an enhanced approach requires a lot of effort for an interim solution. This setup also still suffers from issues with list that commonly update, such as ConfigMaps. Considering those two reasons we have decided not to pursue them and instead accelerate the timeline for Vai as a solution which aims to reach many of the same goals and should be able to handle cases such as constantly changing lists.
